### PR TITLE
refactor: Extract unordered comparison as function

### PR DESCRIPTION
**Pull request by marktwtn**
The extracted functions are refactored to avoid extra variable
declarations.

### CODE REVIEW

File: tests/impl.cpp [hunk 1]
```diff
@@ -350,22 +350,22 @@ result_t do_mm_store_ps(int32_t *p, int32_t x, int32_t y, int32_t z, int32_t w)
...
 {
-    double ret;
+    return (!isnan(a) && !isnan(b)) ? ALL_BIT_1_64 : 0.0f;
+}
 
-    bool isNANA = isnan(a);
-    bool isNANB = isnan(b);
-    ret = (!isNANA && !isNANB) ? ALL_BIT_1_64 : 0.0f;
-    return ret;
+float compunord(float a, float b)
```
Comment (jserv) [inline tests/impl.cpp:361]: Shall we revise the function names for readability?

Comment (marktwtn) [reply tests/impl.cpp:361]: Like `cmpunord` or `existNaNOp`?

Comment (jserv) [reply tests/impl.cpp:361]: We can introduce the name scheme such as `cmp_` (for comparison) `isNaN` (`NaN` may look better than `NAN`).

Comment (jserv) [reply tests/impl.cpp:361]: By the way, this document is useful: https://www.agner.org/optimize/nan_propagation.pdf

Comment (marktwtn) [reply tests/impl.cpp:361]: I will rewrite `compord` as `cmp_noNaN` and `compunord` as `cmp_hasNaN`.

**Pull request merged**